### PR TITLE
Nix improvements

### PR DESCRIPTION
**Pull request by romanofski**

### CODE REVIEW

**Comment by romanofski** [comment]
Don't merge yet. Found a problem with the notmuch binding.

**Comment by romanofski** [comment]
I think it's ready to merge. I wish I could smuggle in a nix-build step in Travis ....

**Comment by romanofski** [comment]
Turned out, making use of Travis to build with nix was easier than I thought.

**Comment by romanofski** [comment]
This is even more interesting than I thought. I guess I won't be able to include a nix test overwriting the compiler to use GHC-8.4.4, since I don't know a way to overwrite cabal in order to use >= 2.4. Instead I overwrite the compiler and use GHC HEAD. Not sure if that is any better tho.

File: test/TestUserAcceptance.hs [hunk 1]
```diff
@@ -939,13 +937,15 @@ sendLiteralKeys keys = do
     waitForString keys defaultCountdown
 
 capture :: ReaderT Env IO String
-capture =
-  tmuxSessionProc "capture-pane"
+capture = T.unpack . decodeLenient . LB.toStrict
```
Comment (frasertweedale) [inline test/TestUserAcceptance.hs]: Just as an FYI, `text` provides functions for decoding lazy bytestrings directly.
I don't think we need bother changing to use it here though (i.e. this shouldn't
be a performance issue here, and even if it was it's in the test suite so I don't
care ^_^)

**Comment by frasertweedale** [comment]
Fine with me.  Merge ho!

File: test/TestUserAcceptance.hs (hunk 1 above)
Comment (romanofski) [reply test/TestUserAcceptance.hs]: Ah cheers! Good to know

**Pull request merged**